src/client/app/helpers/environment.ts

```diff
@@ -295,26 +285,23 @@ class Environment {
 
         const hostInfo = await (async (): Promise<{ host: string, platform: string }> => {
             return new Promise<{ host: string, platform: string }>(async resolve => {
-                let timer = this.createPlaygroundHostReadyTimer();
+                await this.createPlaygroundHostReadyTimer();
+
+                // Now wait for playground host-ready timer to return itself as ready.
+                // If it does, then clear off the previous timeout, and resolve immediately.
 
-                let hostButtonsTimeout = setTimeout(() => {
+                let { host, platform } = Utilities;
```

Comment:
Now that I look at it, it feels like you could just do `const { host } = Utilities`.  And then do the platform bit later, when needed.